amy/workshops/views.py

```diff
@@ -286,7 +286,22 @@ def get_context_data(self, **kwargs):
 
         is_usersocialauth_in_sync = len(self.object.github_usersocialauth) > 0
         context["is_usersocialauth_in_sync"] = is_usersocialauth_in_sync
+        consents = (
+            Consent.objects.filter(person=self.object)
+            .active()
+            .select_related("term", "term_option")
+        )
 
+        context["consents"] = {
+            "May contact": consents.filter(term__slug="may-contact")[0],
+            "Consent to publish profile": consents.filter(term__slug="public-profile")[
```

Comment:
Solution seems a little strange. Happy to put this in the database as an extra field instead.

Reply:
Do we have a way to figure out the term question, e.g. "May contact"?

On a related note, I think using a mapping between `term.slug` and a label ("May contact") should be used to reduce the number of queries.

Reply:
Yes. I'm just not storing a label with the same wording is all. for instance the privacy policy is: *I have read and agree to the data privacy policy of The Carpentries.* I'll revise this. Maybe it's not as big of an issue if they don't map perfectly.